src/d3a/models/market/market_structures.py

```diff
@@ -33,14 +33,17 @@ def my_converter(o):
 
 class Offer:
     def __init__(self, id, time, price, energy, seller,
-                 original_offer_price=None, seller_origin=None):
+                 original_offer_price=None, seller_origin=None, seller_origin_id=None,
+                 seller_id=None):
```

Comment:
May be the higher order objects be better now as arguments for classes and methods with too many arguments:
https://stackoverflow.com/questions/5899185/class-with-too-many-parameters-better-design-strategy

Using `dataclasses ` sounds like a good approach for class/method with too many arguments.

Reply:
I agree. Just by glancing at it, I have the feeling we could benefit from a `Seller` class to store some of those arguments.